compiler/rustc_codegen_ssa/src/traits/backend.rs

```diff
@@ -43,6 +43,8 @@ pub trait CodegenBackend {
 
     fn init(&self, _sess: &Session) {}
 
+    fn name(&self) -> &'static str;
```

Comment:
Can you move this above the `init`? This is not something that needs `init` to be called.